lib/widgets/login.dart

```diff
@@ -235,23 +241,119 @@ class _AddAccountPageState extends State<AddAccountPage> {
   }
 }
 
-class PasswordLoginPage extends StatefulWidget {
-  const PasswordLoginPage({super.key, required this.serverSettings});
+class LoginPage extends StatefulWidget {
+  const LoginPage({super.key, required this.serverSettings});
+
+  /// A key for the page from the last [buildRoute] call.
+  static final _lastBuiltKey = GlobalKey<_LoginPageState>();
+
+  /// The OTP to use, instead of an app-generated one, for testing.
+  @visibleForTesting
+  static String? debugOtpOverride;
+
+  /// Log in using the payload of a web-auth URL like zulip://login?…
+  static Future<void> handleWebAuthUrl(Uri url) async {
+    return _lastBuiltKey.currentState?.handleWebAuthUrl(url);
+  }
 
   final GetServerSettingsResult serverSettings;
 
   static Route<void> buildRoute({required GetServerSettingsResult serverSettings}) {
     return _LoginSequenceRoute(
-      page: PasswordLoginPage(serverSettings: serverSettings));
+      page: LoginPage(serverSettings: serverSettings, key: _lastBuiltKey));
   }
 
   @override
-  State<PasswordLoginPage> createState() => _PasswordLoginPageState();
+  State<LoginPage> createState() => _LoginPageState();
 }
 
-class _PasswordLoginPageState extends State<PasswordLoginPage> {
+class _LoginPageState extends State<LoginPage> {
   bool _inProgress = false;
 
+  String? _otp;
+  String? _getOtp() {
```

Comment:
This is functionally a getter backed by a mutable field, so let's follow the Flutter style guide's order for those:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#order-other-class-members-in-a-way-that-makes-sense

Reply:
Also could make this an actual getter/field pair. E.g. `String? get _otp` and `String? __otp;`.

Reply:
Interesting; yeah, I guess it's fine to have a thing that starts with two underscores. 😛